src/include/catalog/pg_index.h

```diff
@@ -73,6 +73,8 @@ FOREIGN_KEY(indrelid REFERENCES pg_class(oid));
  * ----------------
  */
 typedef FormData_pg_index *Form_pg_index;
+typedef bool (*is_likeam_hook_type)(Oid expected_oid, Oid id);
```

Comment:
typedef bool (* IsIndexAccessMethod_hook_type) (Oid relam, Oid indexAccessMethod);
extern PGDLLIMPORT IsIndexAccessMethod_hook_type IsIndexAccessMethod_hook;

Reply:
‘is’ is a verb, so I think it is more proper with lower-case and refer other hook in cbdb core kernel .

Reply:
Refer to 'planner_hook'、'ProcessUtility_hook'、'ExecutorRun_hook' and other hooks, we should use 'IsIndexAccessMethod_hook_type' instead of 'is_likeam_hook_type', and 'IsIndexAccessMethod_hook' instead of 'is_same_index_am_hook'.

Reply:
The hook's parameters should be the same as the function by type and order, this could avoid logical error.